src/client/helpers/entity.js

```diff
@@ -225,3 +226,61 @@ export function getISBNOfEdition(entity) {
 export function getEditionFormat(entity) {
 	return (entity.editionFormat && entity.editionFormat.label) || '?';
 }
+
+
+/**
+ * Filter the relationship according to relationshipTypeId
+ *
+ * @param {object} entity - Entity with all relationships
+ * @param {number} relationshipTypeId - filter the relationship according to this typeId
+ * @returns {object} retrun the all relationshiops after removing the relatioship with relationshipTypeId = 10
+ */
+export function getFilteredRelationship(entity, relationshipTypeId) {
+	return (Array.isArray(entity.relationships) &&
+				entity.relationships.filter((relation) => relation.typeId !== relationshipTypeId)) || [];
+}
+
+
+/**
+ * Get all works constained by entity according to relationshipTypeId
+ *
+ * @param {object} entity - Edition with all relationships
+ * @param {number} relationshipTypeId - filter the relationship according to this typeId
+ * @returns {object} Return all the works of Edition with relationshipTypeId = 10
+ */
+export function getWorksContainedByEdition(entity, relationshipTypeId) {
```

Comment:
This function and the one below also need renaming, as the refactor has made them reusable for other entity types.

How about `getRelationshipTargetByTypeID` and `getRelationshipSourceByTypeID` ?
Please also review the documentation comments for both methods to reflect the changes.

To finish making it reusable, rename the variables you use in the methods (`works`/`editions` => `target`/`source`).
Instead or returning a full object with bbid and type, return only the filtered array or an empty array

That also means you will need to refactor WorksTable and EditionsTable to take new separate `works` and `editions` property respectively, instead or relying on `entity.works` and `entity.editions`.
(That will result in some silly verbose use for the `PublisherDisplyPage` for example, which will end up like this: `<EditionTable entity={entity} editions={entity.editions}/>`, but so be it, I find it more declarative.)

How does that sound to you?
A bit of a big refactor this late in the PR, but I hadn't grasped how all the pieces worked together.

Reply:
I wanted to add that I'm not asking this for the sake of code purity: I have in mind other display pages where we will want to show certain relationships in their own section like we do here.
I think it makes a lot of sense to show a "main" type of relationship for an entity in a more prominent manner, and users will be expecting that.